examples/remote/client.rs

```diff
@@ -0,0 +1,127 @@
+//! A simple command line client that allows issuing queries to a remote Bevy
+//! app via the BRP.
+
+use std::process;
+
+use anyhow::Result as AnyhowResult;
+use argh::FromArgs;
+use bevy::remote::{DEFAULT_ADDR, DEFAULT_PORT};
+use http_body_util::BodyExt as _;
+use hyper::client::conn::http1;
+use hyper::header::HOST;
+use hyper::Request;
+use macro_rules_attribute::apply;
+use serde_json::Value;
+use smol::{net::TcpStream, Executor};
+use smol_hyper::rt::FuturesIo;
+use smol_macros::main;
+
+/// Struct containing the command-line arguments that can be passed to this example.
+/// The components are passed by their full type names positionally, while `host`
+/// and `port` are optional arguments which should correspond to those used on
+/// the server.
+///
+/// When running this example in conjunction with the `server` example, the `host`
+/// and `port` can be left as their defaults.
+///
+/// For example, to connect to port 1337 on the default IP address and query for entities
+/// with `Transform` components:
+/// ```text
+/// cargo run --example client -- --port 1337 bevy_transform::components::transform::Transform
+/// ```
+#[derive(FromArgs)]
+struct Args {
+    /// the host IP address to connect to
+    #[argh(option, default = "DEFAULT_ADDR.to_string()")]
+    host: String,
+    /// the port to connect to
+    #[argh(option, default = "DEFAULT_PORT")]
+    port: u16,
+    /// the full type names of the components to query for
+    #[argh(positional, greedy)]
+    components: Vec<String>,
+}
+
+/// The application entry point.
+#[apply(main!)]
+async fn main(executor: &Executor<'_>) -> AnyhowResult<()> {
```

Comment:
Is there a strong reason to have this example be this manual and not use something like ureq (or any other client). I feel like most people looking to make a request against the server aren't going to be looking for "how to use `http1::handshake::`", etc, and the async isn't really necessary either.

I put the relevant diff up in a PR for consideration: https://github.com/mweatherley/bevy/pull/2